src/Filter.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Choredo;
+
+class Filter
+{
+    /**
+     * @var string
+     */
+    private $field;
+    /**
+     * @var mixed
+     */
+    private $value;
+    /**
+     * @var callable|null
+     */
+    private $transform;
+
+    /**
+     * Filter constructor.
+     * @param string $field
+     * @param mixed $value
+     * @param callable|null $transform
+     */
+    public function __construct(string $field, $value, callable $transform = null)
+    {
+        $this->field = $field;
+        $this->value = $value;
+        $this->transform = $transform;
+    }
+
+    public function getField(): string
+    {
+        return $this->field;
+    }
+
+    public function getValue()
+    {
+        return is_callable($this->transform) ? call_user_func($this->transform, $this->value) : $this->value;
```

Comment:
1. can you just check `this->transform` since transform can only be callable or null? 
2. can you call transform directly, or do you need `call_user_func`?  `$this->transform($this->value)`

Reply:
I personally prefer `call_user_func`. Looking at `$this->transform()`, it's not immediately apparent whether it's a reference to a instance member containing a callable or if it's calling and instance method. IMO, `call_user_func` is more readable, but very much just a preference ✌️

Reply:
As for `is_callable` vs a boolean check, I feel like the `is_callable` makes it more explicit what's being checked. It's not necessary when reading the code to know that the value can only contain a null or callable. What's relevant in this method is that the behavior changes based on whether or not the value in the variable is explicitly callable. Again, just a readability preference.